test/src/main/java/org/apache/accumulo/test/AuditMessageIT.java

```diff
@@ -124,36 +125,24 @@ private ArrayList<String> getAuditMessages(String stepName) throws IOException {
     // Grab the audit messages
     System.out.println("Start of captured audit messages for step " + stepName);
 
-    ArrayList<String> result = new ArrayList<>();
-    File[] files = getCluster().getConfig().getLogDir().listFiles();
-    assertNotNull(files);
-    for (File file : files) {
-      // We want to grab the files called .out
-      if (file.getName().contains(".out") && file.isFile() && file.canRead()) {
-        try (java.util.Scanner it = new java.util.Scanner(file, UTF_8)) {
-          // strip off prefix, because log4j.properties does
-          final var pattern = Pattern.compile(".* \\["
-              + AuditedSecurityOperation.AUDITLOG.replace("org.apache.", "").replace(".", "[.]")
-              + "\\] .*");
-          while (it.hasNext()) {
-            String line = it.nextLine();
-            if (pattern.matcher(line).matches()) {
-              // Only include the message if startTimestamp is null. or the message occurred after
-              // the startTimestamp value
-              if ((lastAuditTimestamp == null)
-                  || (line.substring(0, 23).compareTo(lastAuditTimestamp) > 0)) {
-                result.add(line);
-              }
+    ArrayList<String> result;
+    final var pattern = Pattern.compile(
+        ".* \\[" + AuditedSecurityOperation.AUDITLOG.replace("org.apache.", "").replace(".", "[.]")
+            + "\\] .*");
+    try (var paths = Files.list(getCluster().getConfig().getLogDir().toPath())) {
+      result = paths.filter(file -> file.getFileName().toString().contains(".out"))
+          .filter(file -> Files.isRegularFile(file) && Files.isReadable(file)).flatMap(path -> {
+            try {
+              return Files.lines(path, UTF_8);
+            } catch (IOException e) {
+              throw new RuntimeException(e);
             }
-          }
-        }
-      }
+          }).filter(pattern.asPredicate())
+          .filter(line -> lastAuditTimestamp == null
+              || line.substring(0, 23).compareTo(lastAuditTimestamp) > 0)
+          .sorted().peek(System.out::println).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Should include comments explaining what this is doing, could probably just include the ones that were there before. Could also condense a couple of the `filter` calls into one.